### PR TITLE
first steps to add sd card support

### PR DESCRIPTION
**Pull request by turbinenreiter**
This adds a new board called 'feather_m0_sd' aka Adafruit M0 Adalogger. The only change so far is the addition of the CS pin for the SD card.
Furthermore, a modified sdcard.py is added to modules.

It's not working tough, probably because SPI isn't working yet.

### CODE REVIEW

**Comment by turbinenreiter** [comment]
Done. The first commit is now useless, should I remove it?

File: atmel-samd/boards/feather_m0_adalogger/conf_usb.h [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+//	(USB_CONFIG_ATTR_REMOTE_WAKEUP|USB_CONFIG_ATTR_SELF_POWERED)
+//	(USB_CONFIG_ATTR_REMOTE_WAKEUP|USB_CONFIG_ATTR_BUS_POWERED)
+
+//! USB Device string definitions (Optional)
+#ifndef USB_DEVICE_MANUFACTURE_NAME
+#  define USB_DEVICE_MANUFACTURE_NAME "Adafruit Industries"
+#endif
+
+#ifndef USB_DEVICE_PRODUCT_NAME
+#  define USB_DEVICE_PRODUCT_NAME "Feather M0 Basic"
```
Comment (tannewt) [inline atmel-samd/boards/feather_m0_adalogger/conf_usb.h]: Change to "Feather M0 Adalogger" please.

**Comment by turbinenreiter** [comment]
Took me a while to figure out how to rebase, but I think it's good now.

File: atmel-samd/boards/feather_m0_adalogger/pins.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
+#ifndef __MICROPY_INCLUDED_ATMEL_SAMD_BOARDS_FEATHER_M0_BASIC_PINS_H__
+#define __MICROPY_INCLUDED_ATMEL_SAMD_BOARDS_FEATHER_M0_BASIC_PINS_H__
```
Comment (tannewt) [inline atmel-samd/boards/feather_m0_adalogger/pins.h]: Oops I missed this the first time. These header guards should be `__MICROPY_INCLUDED_ATMEL_SAMD_BOARDS_FEATHER_M0_ADALOGGER_PINS_H__` to match the filename.

**Comment by turbinenreiter** [comment]
`grep -R basic` now turns up empty.

**Pull request merged**